sdk/ai/azure-ai-inference/azure/ai/inference/tracing.py

```diff
@@ -0,0 +1,676 @@
+# ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import copy
+from enum import Enum
+import functools
+import json
+import importlib
+import logging
+import os
+from typing import Any, Callable, Dict, Iterator, List, Optional, Tuple, Union
+from urllib.parse import urlparse
+
+# pylint: disable = no-name-in-module
+from azure.core import CaseInsensitiveEnumMeta  # type: ignore
+from azure.core.settings import settings
+from . import models as _models
+
+try:
+    # pylint: disable = no-name-in-module
+    from azure.core.tracing import AbstractSpan, SpanKind  # type: ignore
+    from opentelemetry.trace import StatusCode, Span
+
+    _tracing_library_available = True
+except ModuleNotFoundError as _:
+
+    _tracing_library_available = False
+
```

Comment:
the code is not called unless the tracing.py is imported and even if it was, the overhead would be minimal:

I changed the code like so to see the approximate overhead:
**current_time_milliseconds = int(time.time() * 1000)
print("TIME1: " + str(current_time_milliseconds))**
try:
    # pylint: disable = no-name-in-module
    from azure.core.tracing import AbstractSpan, SpanKind  # type: ignore
    from opentelemetry.trace import StatusCode, Span

    _tracing_library_available = True
except ModuleNotFoundError as _:

    _tracing_library_available = False

**current_time_milliseconds = int(time.time() * 1000)
print("TIME2: " + str(current_time_milliseconds))**

and the result was:
TIME1: 1728927982935
TIME2: 1728927982936

Also, i specifically tried to avoid importing inside functions since it would incur overhead every time the function is called, and since function imports are not visible to other functions, having the import just in constructor for example is not possible.